device/transport/amqp/src/amqp.ts

```diff
@@ -682,6 +686,14 @@ export class Amqp extends EventEmitter implements DeviceTransport {
       }
     }
 
+    if (options.hasOwnProperty('productInfo')) {
+      if (typeof(options.productInfo) === 'string') {
+        this._customUserAgentString = options.productInfo;
+      } else {
+        throw new errors.InvalidOperationError('productInfo must be set as a string');
+      }
```

Comment:
typescript guidelines say we should try (within reason) to coerce types. i'd be fine with not testing for string and not throwing.

Reply:
It sounds like you're saying I don't need to check for correct types. Will remove these errors.